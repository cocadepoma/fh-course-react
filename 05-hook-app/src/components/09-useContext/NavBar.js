import React from "react";
import { NavLink } from "react-router-dom";

export const NavBar = () => {
    return (
        <nav className='navbar navbar-expand-lg navbar-dark bg-dark'>
            <div className='container-fluid'>
                <div className='navbar-nav'>
                    <NavLink
                        className='nav-link'
                        activeClassName='active'
                        exact
                        to='/'
                    >
                        Home
                    </NavLink>
                    <NavLink
                        className='nav-link'
                        activeClassName='active'
                        exact
                        to='/about'
                    >
                        About
                    </NavLink>
                    <NavLink
                        className='nav-link'
                        activeClassName='active'
                        exact
                        to='/login'
                    >
                        Login
                    </NavLink>
                </div>
            </div>
        </nav>
    );
};
