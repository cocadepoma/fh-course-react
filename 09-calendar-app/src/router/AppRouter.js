import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import { startChecking } from "../actions/auth";
import { LoginScreen } from "../components/auth/LoginScreen";
import { CalendarScreen } from "../components/calendar/CalendarScreen";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {

    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(startChecking());
    }, [dispatch]);

    const { checking, uid } = useSelector(state => state.auth)

    if (checking) {
        return (<h5>Espere</h5>);
    }

    return (
        <Router>
            <div>
                <Switch>
                    <PrivateRoute exact path='/' isAuthenticated={!!uid} component={CalendarScreen} />
                    <PublicRoute exact path='/login' isAuthenticated={!!uid} component={LoginScreen} />
                    <Redirect to='/' />
                </Switch>
            </div>
        </Router>
    );
};
