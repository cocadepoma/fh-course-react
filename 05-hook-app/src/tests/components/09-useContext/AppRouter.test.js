import { mount } from "enzyme";
import React from "react";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("pruebas en <AppRouter", () => {
    const user = {
        name: "Paco",
        email: "p@peede.com",
    };
    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <AppRouter />
        </UserContext.Provider>
    );
    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
