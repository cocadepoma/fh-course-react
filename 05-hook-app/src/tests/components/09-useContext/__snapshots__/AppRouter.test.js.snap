// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <AppRouter debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Homescreen
      </h1>
      <hr />
      <pre>
        {
   "name": "Paco",
   "email": "p@peede.com"
}
      </pre>
    </div>
  </div>
</div>
`;
